### PR TITLE
services/horizon/ingest: added more description to the history archive metrics labels

### PR DESCRIPTION
**Pull request by sreuland**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

added more content to the history archive request metrics

### Why

there are sub-labels for the history archive metrics key, and the additional docs help describe what each label is counting.

### CODE REVIEW

**Review by jacekn — COMMENTED** [review]
Text looks good but I'm not sure how newline will behave in prometheus.
I only saw single line HELP text. It would be good to test this and confirm how output looks like

**Review by Shaptic — APPROVED** [review]
awesome!!

File: services/horizon/internal/ingest/main.go [hunk 1]
```diff
@@ -409,7 +409,13 @@ func (s *system) initMetrics() {
 	s.metrics.HistoryArchiveStatsCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: "horizon", Subsystem: "ingest", Name: "history_archive_stats_total",
-			Help: "counters of different history archive stats",
+			Help: "Counters of different history archive requests.  " +
+				"'source' label will provide name/address of the physical history archive server from the pool for which a request may be sent.  " +
+				"'type' label will further categorize the potential request into specific requests, " +
+				"'file_downloads' - the count of files downloaded from an archive server, " +
+				"'file_uploads' - the count of files uploaded to an archive server, " +
+				"'requests' - the count of non-download, http query requests sent to an archive server, " +
```
Comment (Shaptic) [inline services/horizon/internal/ingest/main.go:417]: wait isn't requests both downloads + the http queries?

File: services/horizon/internal/ingest/main.go [hunk 2]
```diff
@@ -409,7 +409,13 @@ func (s *system) initMetrics() {
...
+				"'cache_hits' - the count of requests for an archive file that were found on local cache instead, no download request sent to archive server.",
```
Comment (Shaptic) [inline services/horizon/internal/ingest/main.go:418]: and this only grows for downloads that hit, not size checks that hit

**Comment by sreuland** [comment]
> Text looks good but I'm not sure how newline will behave in prometheus. I only saw single line HELP text. It would be good to test this and confirm how output looks like

here's what the final output will look like in metrics output:

```
# HELP horizon_ingest_history_archive_stats_total Counters of different history archive requests.  'source' label will provide name/address of the physical history archive server from the pool for which a request may be sent.  'type' label will further categorize the potential request into specific requests, 'file_downloads' - the count of files downloaded from an archive server, 'file_uploads' - the count of files uploaded to an archive server, 'requests' - the count of all http requests(includes both queries and file downloads) sent to an archive server, 'cache_hits' - the count of requests for an archive file that were found on local cache instead, no download request sent to archive server.
# TYPE horizon_ingest_history_archive_stats_total counter


```

File: services/horizon/internal/ingest/main.go (hunk 1 above)
Comment (sreuland) [reply services/horizon/internal/ingest/main.go]: yes, great catch, corrected - https://github.com/stellar/go/pull/5185/commits/e001d532309f014e158b91b0ab191a50e8118435

**Review by jacekn — APPROVED** [review]
Looks great from my perspective but please get 2nd +1 from somebody with more golang experience

**Pull request merged**